src/classes/PMT_Payment_TEST.cls

```diff
@@ -837,6 +837,92 @@ private class PMT_Payment_TEST {
     }
 
 
+    /**
+     * @description Check that the opportunity amount is updated when the payed payments amounts
+     * exceed the opportunity amount
+     */
+    @isTest
+    private static void testOpportunityAmountSync() {
```

Comment:
Time Permitting, we would have the following tests:

- Tests with Enforce Accounting Data Consistency Enabled:
- Non Finalized Opportunity, Amount changed to more than Sum of Paid Payments: Result (Allowed)
- Non Finalized Opportunity,  Amount changed to less than Sum of Paid Payments: Result (Blocked)
- Finalized Opportunity, Amount same as Sum of Paid Payments: Result (Allowed)
- Finalized Opportunity, Amount < Sum of Paid Payments: Result (Blocked)
- Finalized Opportunity, Amount > Sum of Paid Payments: Result (Blocked)
- Non Final Opportunity inserted with unpaid payments totaling < than opportunity Amount.  Change Unpaid Payments to Paid.  Amount should remain the same
- Non Finalized Opportunity inserted with unpaid payments totaling > than opportunity Amount. Change Unpaid Payments to Paid.  Amount should increase to Sum of Paid Payments
- Non Finalized Opportunity inserted with unpaid payments totaling < Opportunity Amount.  Change Unpaid Payments to Paid.  No change to Opportunity Amount.  Change Opportunity to Finalized Stage.  Amount should change to Sum of Paid Payments-- I think as coded, this will instead throw a validation error, as it does not re-sum the Payment Amounts when going to finalized though). @pedrodalcol - Do I have the behavior we want correct here?  Basically, when an Opportunity Becomes finalized (but was not previously, it needs to recalculate the Amount based on the Paid Payments.  The logic as coded currently will not do that because we are only recalcing on payment changes..
- Non Finalized Opportunity inserted with unpaid payments.  Change to Finalized Opportunity.  Update the payments to Paid.  Confirm Opportunity Amount is now Sum of Paid Payments:  Note: I don't love this test because it will eventually conflict with write off on finalized behavior.
- Finalized Opportunity with Paid Payment equal to amount.  Add additional unpaid payment.  No change in  Opp amount.  Add additional paid payment confirm Opp AMount increases.
- Finalized Opportunity with Paid Payments equal to Amount.  Change one of the Payments to Unpaid or Written Off.  Confirm Opportunity Amount updates.

-- Tests with Enforce Accounting Data Consistency Not Enabled:
- Create Non Finalized Opportunity with Amount > Paid Payments.  Change Opp Amount to < Sum of Paid Payment Amounts.  Confirm no error.
- Create Non Finalized Opportunity with Amount < Unpaid Payments.  Make Payments Paid.  Confirm No change to Opp Amount.
- Create Finalized Opportunity With Amount Different than Paid Payments.  Confirm no error.   Update Amount.  Confirm no error.
- Create Finalized Opportunity With Amount and Unpaid Payments.  Change Unpaid Payments to Paid.  Confirm no change in amount.

I realize you will not be able to get to these in the time you have.  Can you add comments with a //TODO: to this so we can add these tests later?